packages/network-controller/src/NetworkController.ts

```diff
@@ -42,6 +43,25 @@ export type NetworkProperties = {
   isEIP1559Compatible?: boolean;
 };
 
+/**
+ * Custom RPC network information
+ *
+ * @property rpcUrl - RPC target URL.
+ * @property chainId - Network ID as per EIP-155
+ * @property nickname - Personalized network name.
+ * @property ticker - Currency ticker.
+ * @property rpcPrefs - Personalized preferences.
+ */
+export type NetworkConfiguration = {
+  rpcUrl: string;
+  chainId?: string;
```

Comment:
I wonder if we should take this time to make this a little more strict. Is there a case where we'd really want the `chainId` to be optional?

Reply:
Ah, I see. I guess that it makes sense that this would be optional given that a chain already knows its chain ID.

Reply:
I think we need to make this mandatory for mobile/extension compatibility reasons though, so it's still a good point. I have added an issue here so that we don't forget: https://github.com/MetaMask/core/issues/1078

I believe it's already mandatory on mobile but I need to confirm.

Reply:
Yeah I believe this already mandatory on both apps. Will double check though

Reply:
Mobile custom network form requires it: ![IMG_DF107F86986B-1](https://user-images.githubusercontent.com/34557516/214648671-b3f0e1f1-d3d2-4016-9672-4c2c0d9c0f79.jpeg)
Extension custom network form requires it (submit button is disabled unless it's added:
![Screenshot 2023-01-25 at 12 19 40 PM](https://user-images.githubusercontent.com/34557516/214649103-cdbc1f43-dd8a-41c5-b8a3-b51f56c18d90.png)
Extension RPC method `wallet_watchEthereumChain` requires it:
![Screenshot 2023-01-25 at 12 21 04 PM](https://user-images.githubusercontent.com/34557516/214649379-f2337b9d-eb32-40fe-8379-35e99fa02dce.png)

only one I'm not totally sure about is mobiles implementation of `wallet_watchEthereumChain`... I'd assume it does, but can ask/check

Reply:
@adonesky1 What is the status on this? Do we need to still check mobile or do we want to go ahead and merge this with `chainId` being optional and come back later and change this?

Reply:
Looping back to confirm this now!

Reply:
Looks to me like mobile implementation of `wallet_addEthereumChain` will throw [here](https://github.com/MetaMask/metamask-mobile/blob/main/app/core/RPCMethods/wallet_addEthereumChain.js#L96) if no `chainId` is provided. So I believe we can make `chainId` a required value here

Reply:
cc @mcmire ☝️

Reply:
Nice!

Reply:
Ah I didn't notice earlier that Mark had made an issue for this. I guess we can leave this alone for now and address it in another commit to complete that issue.